fitbenchmarking/core/fitting_benchmarking.py

```diff
@@ -65,37 +128,204 @@ def fitbenchmark_group(group_name, options, data_dir):
-               sorted(results_dict.keys(), key=str.lower)]
+        controller.parameter_set = start_values_index
 
-    return results, failed_problems, unselected_minimzers
+        #########################
+        # Loops over minimizers #
+        #########################
+        problem_result, minimizer_failed = \
+            loop_over_minimizers(controller=controller,
+                                 minimizers=minimizers,
+                                 options=options)
+        unselected_minimzers[s] = minimizer_failed
+        software_results.extend(problem_result)
+
+    # Checks to see if all of the minimizers raise and exception and
+    # records the problems name for that case
+    software_check = [np.isinf(v.chi_sq) for v in software_results]
+    if all(software_check):
+        software_results = []
+        problem_fails.append(problem.name)
+    results.extend(software_results)
+
+    return results, problem_fails, unselected_minimzers
+
+
+def loop_over_minimizers(controller, minimizers, options):
+    """
+    Loops over minimizers in fitting software
+
+    :param controller: The software controller for the fitting
+    :type controller: Object derived from BaseSoftwareController
+    :param minimizers: array of minimizers used in fitting
+    :type minimizers: list
+    :param options: FitBenchmarking options for current run
+    :type options: fitbenchmarking.utils.options.Options
+
+    :return: list of all results and dictionary of unselected minimizers
+             based on algorithm_type
+    :rtype: tuple(list of fibenchmarking.utils.fitbm_result.FittingResult,
+                  list of failed minimizers)
+    """
+    grabbed_output = output_grabber.OutputGrabber(options)
+    problem = controller.problem
+    jac = controller.problem.jac
+
+    results_problem = []
+    minimizer_failed = []
+
+    num_runs = options.num_runs
+    algorithm_type = options.algorithm_type
+
+    for minimizer in minimizers:
+        minimizer_check = True
+        LOGGER.info("            Minimizer: %s", minimizer)
+
+        controller.minimizer = minimizer
+        try:
+            with grabbed_output:
+                controller.validate_minimizer(minimizer, algorithm_type)
+                # Calls timeit repeat with repeat = num_runs and number = 1
+                runtime_list = \
+                    timeit.Timer(setup=controller.prepare,
+                                 stmt=controller.fit).repeat(num_runs, 1)
+                runtime = sum(runtime_list) / num_runs
+                controller.cleanup()
+        # Catching all exceptions as this means runtime cannot be calculated
+        # pylint: disable=broad-except
+        except Exception as excp:
+            if isinstance(excp, UnknownMinimizerError):
+                minimizer_failed.append(minimizer)
+                minimizer_check = False
+            LOGGER.warn(str(excp))
+
+            runtime = np.inf
+            controller.flag = 3
```

Comment:
Unrelated to this issue I think. How well do we currently describe the different flag values? Do we document this beyond `def check_attributes(self):` in base_controller.py?

Reply:
I agree with Anders, this doesn't appear to be documented anywhere - I think you have to look at the code in fitbenchmark_one_problem.py to determine what the flags mean.

Reply:
The documentation is [here](https://fitbenchmarking.readthedocs.io/en/latest/extending/controllers.html) for when you add a new controller. Do you think we should add more somewhere else?

Reply:
I thought it would be almost as quick to solve the problem as explain my suggested solution, so I've created an issue and a PR - no problems if you want to go down a different route for this.

Reply:
I have just looked over this PR and I agree that this is a much nicer way of doing it